### PR TITLE
Phase2-hgx85 Correct for overwriting (as reported in PR #19179)

### PR DESCRIPTION
**Pull request by bsunanda**
Use correct # for maximum wafers. Working on getting some of the magic #'s from geometry & topology/

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+1

**Comment by kpedro88** [comment]
@bsunanda can you make a backport of this fix for 91X?

File: RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h [hunk 1]
```diff
@@ -164,12 +164,11 @@ static const unsigned int maxlayer = 52;
 
 
 private:
-
 // last layer per subdetector
 static const unsigned int lastLayerEE = 28;
 static const unsigned int lastLayerFH = 40;
 // maximum number of wafers per Layer: 666 (V7), 794 (V8)
-static const unsigned int maxNumberOfWafersPerLayer = 794;
+static const unsigned int maxNumberOfWafersPerLayer = 796;
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h]: Please update the text in the comments above: it still says 794. If instead this 796 corresponds to 794 precisely, please add a few words why.

File: RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc [hunk 2]
```diff
@@ -568,8 +568,8 @@ void HGCalImagingAlgo::computeThreshold() {
 	  else if( thickness>299. && thickness<301. ) thickIndex=2;
 	  else assert( thickIndex>0 && "ERROR - silicon thickness has a nonsensical value" );
 	  float sigmaNoise = 0.001 * fcPerEle * nonAgedNoises[thickIndex] * dEdXweights[layer] / (fcPerMip[thickIndex] * thicknessCorrection[thickIndex]);
-	  thresholds[layer-1][wafer]=sigmaNoise*ecut;
-	  v_sigmaNoise[layer-1][wafer] = sigmaNoise;
+	  thresholds[layer-1].at(wafer)=sigmaNoise*ecut;
+	  v_sigmaNoise[layer-1].at(wafer) = sigmaNoise;
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc]: now that the problem was fixed, please change back to the faster "[]" version

Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc]: if protection is really needed for out-of bound reads, please add an ```assert(wafer < maxNumberOfWafersPerLayer)``` on L558.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**